<a href="https://colab.research.google.com/github/sdasmaha/Ml_Engineers_445/blob/master/Exercise_2_Notebook_SrinjoyDasMahapatra.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Notebook Excercise #2
#Srinjoy DasMahapatra

IDE => Google Colab

In [10]:
import numpy as np
import pandas as pd
from scipy import stats as sps
from matplotlib import pyplot as plt
from mpl_toolkits.mplot3d import Axes3D
from IPython.display import display, Latex


In [12]:
from google.colab import drive
drive.mount("/content/gdrive")

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [13]:
df_iot_tel = pd.read_csv("/content/gdrive/My Drive/Colab Notebooks/BAJWA-ML/Notebook 2/iot_telemetry_dataset.csv")

# 1. Feature Engineering for _Environmental Sensor Telemetry Data_

## Problem 1.1

###(a)

From Videos ML

### (b)

In [14]:
df_iot_tel.head(5)

ts             device        co  ...  motion     smoke       temp
0  1.594512e+09  b8:27:eb:bf:9d:51  0.004956  ...   False  0.020411  22.700000
1  1.594512e+09  00:0f:00:70:91:0a  0.002840  ...   False  0.013275  19.700001
2  1.594512e+09  b8:27:eb:bf:9d:51  0.004976  ...   False  0.020475  22.600000
3  1.594512e+09  1c:bf:ce:15:ec:4d  0.004403  ...   False  0.018628  27.000000
4  1.594512e+09  b8:27:eb:bf:9d:51  0.004967  ...   False  0.020448  22.600000

[5 rows x 9 columns]

In [15]:
print(df_iot_tel.shape) 

(405184, 9)


There are **405,184** data samples in the dataset.

### (c)



In [21]:
df_iot_tel['device'].dtypes

dtype('O')

We can see the MAC address is of datatype=> **O** (Object)

In [22]:
df_iot_tel['device'].where(df_iot_tel['device'] == '00:0f:00:70:91:0a').value_counts()

00:0f:00:70:91:0a    111815
Name: device, dtype: int64

There are **111,815** data samples with the MAC address 00:0f:00:70:91:0a 

### (d)

In [23]:
df_iot_tel['device'].where(df_iot_tel['device'] == '1c:bf:ce:15:ec:4d').value_counts()

1c:bf:ce:15:ec:4d    105918
Name: device, dtype: int64

There are **105,918** data samples with the MAC address 1c:bf:ce:15:ec:4d


### (e)

In [24]:
df_iot_tel['device'].where(df_iot_tel['device'] == 'b8:27:eb:bf:9d:51').value_counts()

b8:27:eb:bf:9d:51    187451
Name: device, dtype: int64

There are **187,451** data samples with the MAC address b8:27:eb:bf:9d:51

**Adding all the MAC addresses individually we get 405184.  Hence there is only 3 MAC addresses in total**

## Problem 1.2

### (a)

 The problem mentions most of the challenges that otherwise needed to be worked on : 
 1.  No missing Entries
 2. pandas.get_dummies() can be used 


**DEVICE**

In [28]:
MAC_types = (' 00:0f:00:70:91:0a ',' 1c:bf:ce:15:ec:4d ',' b8:27:eb:bf:9d:51 ')
MAC_df = pd.DataFrame(MAC_types, columns=['MAC_df'])
# generate binary values using get_dummies
dum_df = pd.get_dummies(MAC_df, columns=["MAC_df"], prefix=["MAC:"] )
# merge with main df bridge_df on key values
MAC_df = MAC_df.join(dum_df)
MAC_df

MAC_df  ...  MAC:_ b8:27:eb:bf:9d:51 
0   00:0f:00:70:91:0a   ...                         0
1   1c:bf:ce:15:ec:4d   ...                         0
2   b8:27:eb:bf:9d:51   ...                         1

[3 rows x 4 columns]

**Light**

In [ ]:
# your code for 1.2a goes here

### (b)

In [ ]:
# your code for 1.2b goes here

### (c)

In [ ]:
# your code for 1.2c goes here

### (d)

_Your answer for 1.2d goes here_

### (e)

In [ ]:
# your code for 1.2e goes here

## Problem 1.3

In [ ]:
# Code of 1.3 goes here

# 2. Feature Learning for _Synthetically Generated Data_

## Problem 2.1

### (a)

In [ ]:
# your code for 2.1a

### (b)

In [ ]:
# your code for 2.1b

### (c)

In [ ]:
# your code for 2.1c

### (d)

In [ ]:
# your code for 2.1d

## Problem 2.2

In [ ]:
# please structure your code and markdown cells in a similar fashion to 2.1

## Problem 2.3

In [ ]:
# please structure your code and markdown cells in a similar fashion to 2.1

## Problem 2.4


In [ ]:
# please structure your code and markdown cells in a similar fashion to 2.1